abci/types/application.go

```diff
@@ -96,7 +96,7 @@ func (BaseApplication) ApplySnapshotChunk(req RequestApplySnapshotChunk) Respons
 
 func (BaseApplication) PrepareProposal(req RequestPrepareProposal) ResponsePrepareProposal {
 	// we use placeholder values for the hash
-	req.Txs = append(req.Txs, tmhash.Sum(nil))
+	req.Txs = append(req.Txs, merkle.HashFromByteSlices(req.Txs))
```

Comment:
This is what was causing the test to fail. We were using idential hashes so the blockID.Hash was the same but blockID.PartSetHeader was different